src/components/MyAccount/RequestsTab/UpdateLocation.tsx

```diff
@@ -0,0 +1,196 @@
+}
+
+const UpdateLocation = ({
+  updateHoldLocation,
+  pickupLocationOptions,
+  patronId,
+  holdId,
+  pickupLocation,
+  key,
+}: UpdateLocationPropsType) => {
+  const { Modal, onOpen: openModal, onClose: closeModal } = useModal()
+  const locationsWithSelectedFirst = [
+    pickupLocation,
+    ...pickupLocationOptions.filter(
+      (loc) => loc.code.trim() !== pickupLocation.code.trim()
+    ),
+  ]
+  const confirmLocationChangeModalProps = (selected: SierraCodeName) => ({
+    type: "confirmation",
+    bodyContent: (
+      <Box className={styles.modalBody}>
+        <Select
+          value={selected.code}
+          onChange={(e: React.FormEvent<HTMLInputElement>) => {
+            const newLocation = locationsWithSelectedFirst.find(
+              (loc) => e.currentTarget.value === loc.code
+            )
+            setModalProps(
+              confirmLocationChangeModalProps(
+                newLocation
+              ) as ConfirmationModalProps
+            )
+          }}
+          id={`update-location-selector-${key}`}
+          labelText="Pickup location"
+          showLabel
+          name={"pickup-location"}
+        >
+          {locationsWithSelectedFirst.map((location, locationKey) => {
+            return (
+              <option
+                key={`pickupLocation-option-${key}${locationKey}`}
+                value={location.code}
+              >
+                {location.name}
+              </option>
+            )
+          })}
+        </Select>
+      </Box>
+    ),
+    closeButtonLabel: "Cancel location change",
+    confirmButtonLabel: "Confirm location",
+    onCancel: () => {
+      closeModal()
+    },
+    onConfirm: async () => {
+      setModalProps({
+        ...confirmLocationChangeModalProps(selected),
+        bodyContent: <SkeletonLoader showImage={false} />,
+      } as ConfirmationModalProps)
+      const response = await fetch(
+        `${BASE_URL}/api/account/holds/update/${holdId}`,
+        {
+          method: "PUT",
+          body: JSON.stringify({
+            pickupLocation: selected.code,
+            patronId: `${patronId}`,
+          }),
+        }
+      )
+      if (response.status == 200) {
+        // Open next modal to confirm request has been canceled.
+        setModalProps(successModalProps(selected) as DefaultModalProps)
+      } else setModalProps(failureModalProps as DefaultModalProps)
+    },
+    headingText: (
+      <Heading className={styles.modalHeading}>
+        <Text sx={{ marginBottom: 0 }}>
+          {" "}
+          Where would you like to pick up this item?{" "}
+        </Text>
+      </Heading>
+    ),
+  })
+
+  const [modalProps, setModalProps] = useState<BaseModalProps>(
+    confirmLocationChangeModalProps(pickupLocation) as ConfirmationModalProps
+  )
+
+  const successModalProps = (newLocation: SierraCodeName) => ({
+    type: "default",
+    bodyContent: (
+      <Box className={styles.modalBody}>
+        <Text sx={{ marginLeft: "l" }}>
```

Comment:
Very small nit, i slightly prefer the margin props ("ml", "mb" etc) since they're available and a bit more concise but i'll leave that up to you.